pkg/target/vapp_dto.go

```diff
@@ -48,61 +78,36 @@ func (vapp *VirtualApp) getCommoditiesBought(vAppBuilder *builder.EntityDTOBuild
 			}
 
 			app := container.App
-			appCommodity, err := builder.NewCommodityDTOBuilder(proto.CommodityDTO_TRANSACTION).
-				Key(app.UUID).
-				Used(app.QPS.Used).
-				Create()
+			appCommodity, err := vapp.createAppCommodity(pod, container, proto.CommodityDTO_TRANSACTION,
+				app.QPS.Used, app.QPS.Capacity)
 			if err != nil {
-				glog.Errorf("failed to create commodity bought for VirtualApp[%s]-pod[%s]-container[%s]-app[%s]",
-					vapp.Name, pod.Name, container.Name, app.Name)
 				continue
 			}
-
+			qpsused += app.QPS.Used
+			qpscap += app.QPS.Capacity
 			bought := []*proto.CommodityDTO{appCommodity}
 
+			if app.ResponseTime != nil {
```

Comment:
In supplyChain definition, VApp will buy `ResponseTime` from `Application`. If it does not buy, then the supplyChain check may fail.

Reply:
I'm not so sure about this.  I believe I've tested this with and without adding the ResponseTime commodity, and I've seen no errors.  I will re-check this, and if required, I will include a ResponseTime commodity with used/capacity of zero.

Reply:
After discussing this with Songbin, we are now going to require that the response time and transaction (QPS) values always be specified.  I will remove any logic that attempts to accommodate these missing parameters and will flag a parsing error when they are missing.